src/main/scala/chisel3/simulator/scalatest/HasCliOptions.scala

```diff
@@ -0,0 +1,166 @@
+
+  import HasCliArguments._
+
+  private val options = mutable.HashMap.empty[String, CliOption[_]]
+
+  final def addOption(option: CliOption[_]): Unit = {
+    if (options.contains(option.name))
+      throw new Exception("unable to add option with name '$name' because this is already taken by another option")
+
+    options += option.name -> option
+  }
+
+  private def helpBody = {
+    val optionsHelp = options.map { case (_, option) =>
+      s"""|  ${option.name}
+          |      ${option.help}
+          |""".stripMargin
+    }.mkString
+    s"""|Usage: <ScalaTest> [-D<name>=<value>...]
+        |
+        |This ChiselSim ScalaTest test supports passing command line arguments via
+        |ScalaTest's "config map" feature.  To access this, append `-D<name>=<value>` for
+        |a legal option listed below.
+        |
+        |Options:
+        |
+        |$optionsHelp""".stripMargin
+  }
+
+  private def illegalOptionCheck(): Unit = {
+    configMap.keys.foreach { case name =>
+      if (!options.contains(name)) {
+        throw new IllegalArgumentException(
+          dramaticMessage(
+            header = Some(s"illegal ChiselSim ScalaTest option '$name'"),
+            body = helpBody
+          )
+        ) with NoStackTrace
+      }
+    }
+  }
+
+  implicit def commonSettingsModifications: svsim.CommonSettingsModifications = {
+    illegalOptionCheck()
+    options.values.foldLeft(_: CommonCompilationSettings) { case (acc, option) =>
+      configMap.getOptional[String](option.name) match {
+        case None => acc
+        case Some(value) =>
+          option.updateCommonSettings.apply(option.convert(value), acc)
+      }
+    }
+  }
+
+  implicit def backendSettingsModifications: svsim.BackendSettingsModifications = {
+    illegalOptionCheck()
+    options.values.foldLeft(_: Backend.Settings) { case (acc, option) =>
+      configMap.getOptional[String](option.name) match {
+        case None => acc
+        case Some(value) =>
+          option.updateBackendSettings.apply(option.convert(value), acc)
+      }
+    }
+  }
+
+  addOption(
+    CliOption[Unit](
+      name = "help",
+      help = "display this help text",
+      convert = _ => {
+        throw new IllegalArgumentException(
+          dramaticMessage(
+            header = Some("help text requested"),
+            body = helpBody
+          )
+        ) with NoStackTrace
+      },
+      updateCommonSettings = (_, a) => a,
+      updateBackendSettings = (_, a) => a
+    )
+  )
+
+}
+
+object CLI {
+
+  import HasCliArguments.CliOption
+
+  trait VcdCapability { this: HasCliArguments =>
+
+    addOption(
+      CliOption[Unit](
+        name = "withVcdCapability",
+        help = "compiles the simulator with VCD support. (Use `enableWaves` to dump a VCD.)",
+        convert = value => {
+          val trueValue = Set("true", "1")
+          trueValue.contains(value) match {
+            case true => {}
```

Comment:
```suggestion
            case true => ()
```
Nit but I much prefer `()` as the explicit `Unit`.

Reply:
Fixed in: b4fc002ec